### PR TITLE
New

### PR DESCRIPTION
**Pull request by VadimDrobyazko**
[DEMO LINK](https://vadimdrobyazko.github.io/react_people-table-basics/)

Я не понимаю почему не работают ссылки mother, father. Help

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,173 +1,120 @@
...
+
+export const App: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+
+  const getpeople = async () => {
+    const fetchPeople = await getPeople();
+
+    setIsLoading(true);
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:43]:
```suggestion
    setIsLoading(true);
    
    const fetchPeople = await getPeople();
```
it should fix this test
Screenshot from 2023-05-13 16-30-49

File: src/App.tsx [hunk 2]
```diff
@@ -1,173 +1,120 @@
...
+
+    try {
+      setPeople(fetchPeople);
+    } catch {
+      setIsLoading(false);
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:48]:
```suggestion
```
redundant 
finally block will word in any case

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]

File: src/components/PeoplePage/PeoplePage.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+import { Person } from '../../types/Person';
+
+type Props = {
+  people: Person[];
+  error: boolean,
```
Comment (etojeDenys) [inline src/components/PeoplePage/PeoplePage.tsx:7]: Missing semicolon at the end of the line.

File: src/components/PeopleTable/PeopleTable.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+  return (
+    <>
+      {error && (
+        <p data-cy="peopleLoadingError" className="has-text-danger">
+          Something went wrong
+        </p>
+      )}
+
+      {people.length < 0 ? (
```
Comment (etojeDenys) [inline src/components/PeopleTable/PeopleTable.tsx:23]: Replace 'people.length < 0' with 'people.length === 0' to correctly check for an empty array of people.

File: src/App.tsx [hunk 5]
```diff
@@ -1,173 +1,119 @@
...
+
+export const App: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+
+  const getpeople = async () => {
+    setIsLoading(true);
+
+    const fetchPeople = await getPeople();
```
Comment (etojeDenys) [inline src/App.tsx:43]: Missing error handling for the API call. Wrap the API call in a try-catch block.

File: src/App.tsx [hunk 6]
```diff
@@ -1,173 +1,119 @@
...
+    </>
+  );
+};
+
+export const App: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+
+  const getpeople = async () => {
```
Comment (etojeDenys) [inline src/App.tsx:40]: The function name `getpeople` should be in camelCase as `getPeople`.

File: src/App.tsx [hunk 7]
```diff
@@ -1,173 +1,119 @@
...
+import { PeoplePage } from './components/PeoplePage/PeoplePage';
+
+type Props = {
+  to: string;
+  text: string;
+};
+
+export const PageNavLink: React.FC<Props> = ({ to, text }) => {
+  return (
+    <>
```
Comment (etojeDenys) [inline src/App.tsx:21]: redundant fragment

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**